src/algebra/module/submodule_lattice.lean

```diff
@@ -157,6 +158,33 @@ lemma mem_Sup_of_mem {S : set (submodule R M)} {s : submodule R M}
   (hs : s ∈ S) : ∀ {x : M}, x ∈ s → x ∈ Sup S :=
 show s ≤ Sup S, from le_Sup hs
 
+/-- This is the statement that `coe : (submodule R M) → set M` is Scott continuous for the
+ω-complete partial order induced by the complete lattice structures. See
+`omega_complete_partial_order.continuous`. -/
+@[norm_cast, simp] lemma coe_supr_of_chain_eq_supr_coe (a : ℕ →ₘ submodule R M) :
+  (↑(⨆ k, a k) : set M) = ⋃ k, (a k : set M) :=
+begin
+  apply le_antisymm,
+  { let p : submodule R M :=
+    { carrier   := ⋃ k, (a k : set M),
+      zero_mem' := by { rw set.mem_Union, use 0, simp only [set_like.mem_coe, submodule.zero_mem] },
+      add_mem'  := λ x y hx hy, by
+      { rw set.mem_Union at ⊢ hx hy, obtain ⟨i, hi⟩ := hx, obtain ⟨j, hj⟩ := hy,
+        cases le_total i j with h; replace h := a.monotone h; rw set_like.le_def at h,
+        { use j, exact (a j).add_mem (h hi) hj, },
+        { use i, exact (a i).add_mem hi (h hj), }, },
+      smul_mem' := λ r x hx, by
+      { rw set.mem_Union at ⊢ hx, obtain ⟨i, hi⟩ := hx, use i, simp only [set_like.mem_coe],
+        exact (a i).smul_mem r hi, }, },
+    change _ ⊆ ↑p, rw set_like.coe_subset_coe,
+    exact supr_le (set.subset_Union _), },
+  { suffices : monotone (coe : submodule R M → set M), { exact this.le_map_supr, },
+    intros p q hpq, simp only [set_like.coe_subset_coe, set.le_eq_subset, hpq], },
+end
```

Comment:
After moving to `linear_algebra.basic`, this is provable as
```suggestion
submodule.coe_supr_of_directed a $ directed_of_sup a.monotone
```

Reply:
I think it might sense to add an alias `monotone.directed` to `directed_of_sup` just to enable dot notation and discoverability here.

Reply:
Much better :-)